ConfigSpace/io/pcs_new.py

```diff
@@ -347,16 +354,40 @@ def read(pcs_string, debug=False):
                         condition = LessThanCondition(child, parent, restrictions)
                     elif operation == '>':
                         condition = GreaterThanCondition(child, parent, restrictions)
-        
-                    condition_objects.append(condition)       
+            
+                    condition_objects.append(condition)
+
         if len(condition_objects) > 1:
-            for connective in connectives:
-                if connective == '&&':
-                    and_conjunction = AndConjunction(*condition_objects)
-                    configuration_space.add_condition(and_conjunction)
-                elif connective == '||':
+            if '||' in connection:
+                ands = []
+                ors = []
+                if '&&' in connection:
+                    if (len(connection) % 2 != 0) and (connection.count('&&') % 2 == 0):
+                        i = 0
+                        j = 0
+                    else:        
+                        i = 1
+                        j = 0
+                    while i < len(condition_objects):
+                        while j < len(connection):
+                            if connection[j] == '&&':
+                                ands.append(condition_objects[i-1])
+                                ands.append(condition_objects[i])
+                                i += 2
+                                j += 1
+                            else:
+                                ors.append(condition_objects[i-1])
+                                i += 1
+                                j += 1
+
+                    orand_conjunction = OrConjunction(AndConjunction(*ands), *ors)
```

Comment:
This only creates a single and conjunction inside an or conjunction. Instead, you want to connect multiple and conjunctions inside an or conjunction. Also, I think the logic of this code can be simplified. How about:
```
ands = []
ors = []
for i, conn in enumerate(connections):
    if conn == '||':
        ands.append(condition_objects[i])
        if len(ands) > 1:
            and = AndConjunction(*ands)
        else:
            and = ands[0]
        ors.append(and)
        ands = []
# Add the final object as well
ands.append(condition_objects[-1]
if len(ands) > 0:
    if len(ands) > 1:
        and = AndConjunction(*ands)
    else:
        and = ands[0]
    ors.append(and)
if len(ors) > 1:
    final_conjunction = OrConjunction(*ors)
else:
    final_conjunction = ors[0]
```
Also, I that there is no unit test for this piece of code. It would be good if you first write a few unit tests for several conditions that happen in practive.
```